auacm/app/static/js/sockets.js

```diff
@@ -0,0 +1,14 @@
+function Socket(url) {
+    this.ws = new WebSocket(url);
+    this.events = {};
+    var parent = this;
+    this.on = function(event, func) {
+        parent.events[event] = func;
+    };
+    this.ws.onmessage = function(event) {
+        var data = JSON.parse(event.data);
+        if (data.eventType in parent.events) {
+            parent.events[data.eventType](data.data);
+        }
+    };
+}
```

Comment:
Yeah, if we do any sort of client communication to the server via sockets, then this will need to be expanded upon. I did just notice, however that I shouldn't be setting some variables to `this.*`. They should be function scope only for encapsulation.